### PR TITLE
add scale property permitting to modify pixels per point value

### PR DESCRIPTION
**Pull request by buxx**
Hello,

This PR is relative to this issue : https://github.com/not-fl3/egui-miniquad/issues/39.

I would like to permit modify pixels_per_point value (currently reset at each frame). I would be happy to receive your review on this PR (I don't know if my approach is good with your architecture or philosophy).

Thanks for your work, 
buxx.

### CODE REVIEW

**Comment by emilk** [comment]
I'm not sure this is the right approach. We already have `dpi_scale` and `pixels_per_point`, and this PR introduces a third `scale`. I think it would be more natural to just allow modifying `pixels_per_point`. The current problem is that it is being reset each frame: https://github.com/not-fl3/egui-miniquad/blob/master/src/input.rs#L12 – so let's fix that instead. Then a user can do `egui_ctx.set_pixels_per_point(…)` and it will just work. And if they want to reset it to the system default, they can just do `egui_ctx.set_pixels_per_point(mq_ctx.dpi_scale());`

**Comment by buxx** [comment]
@emilk I agree with idea about to not add a third `scale`. I thirstly on permited modification of `egui_input.pixels_per_point = Some(pixels_per_point);` but rendering is not impacted because `paint` function compute screen size, which care about screen size and dpi scale (without taking care about pixels per points) : 



(Then, I use `scale` property about recognize real mouse position)

I think my code can be simplified anyway. Maybe, instead of use `scale` property, value of egui context pixels per point can be used ?

File: examples/demo.rs [hunk 1]
```diff
@@ -22,6 +24,7 @@ impl mq::EventHandler for Stage {
     fn update(&mut self, _ctx: &mut mq::Context) {}
 
     fn draw(&mut self, ctx: &mut mq::Context) {
+        self.egui_mq.egui_input().pixels_per_point = Some(self.scale.min(2.0).max(0.7));
```
Comment (emilk) [inline examples/demo.rs:27]:
```suggestion
        self.egui_mq.egui_input().pixels_per_point = Some(self.scale.clamp(0.7, 2.0));
```

…but it is better to do the clamping when you modify the scale

File: src/lib.rs [hunk 2]
```diff
@@ -192,17 +197,24 @@ impl EguiMq {
...
             self.textures_delta.clear();
         } else {
             eprintln!("Failed to draw egui. You need to call `end_frame` before calling `draw`");
         }
     }
 
     /// Call from your [`miniquad::EventHandler`].
-    pub fn mouse_motion_event(&mut self, ctx: &mut mq::Context, x: f32, y: f32) {
-        let pos = egui::pos2(x as f32 / ctx.dpi_scale(), y as f32 / ctx.dpi_scale());
+    pub fn mouse_motion_event(&mut self, _: &mut mq::Context, x: f32, y: f32) {
```
Comment (emilk) [inline src/lib.rs:213]:
```suggestion
    pub fn mouse_motion_event(&mut self, x: f32, y: f32) {
```

File: src/lib.rs [hunk 3]
```diff
@@ -227,12 +239,15 @@ impl EguiMq {
     /// Call from your [`miniquad::EventHandler`].
     pub fn mouse_button_down_event(
         &mut self,
-        ctx: &mut mq::Context,
+        _: &mut mq::Context,
```
Comment (emilk) [inline src/lib.rs:242]: rm

File: src/lib.rs [hunk 4]
```diff
@@ -245,12 +260,15 @@ impl EguiMq {
     /// Call from your [`miniquad::EventHandler`].
     pub fn mouse_button_up_event(
         &mut self,
-        ctx: &mut mq::Context,
+        _: &mut mq::Context,
```
Comment (emilk) [inline src/lib.rs:263]: rm

**Review by emilk — CHANGES_REQUESTED** [review]

**Review by emilk — CHANGES_REQUESTED** [review]
This is definitely a step in the right direction (decoupling `dpi_scale` from `pixels_per_point` and using the correct thing in the correct places).

Left some nits though

File: src/lib.rs (hunk 4 above)
Comment (buxx) [reply src/lib.rs:263]: done

File: src/lib.rs (hunk 3 above)
Comment (buxx) [reply src/lib.rs:242]: done

File: src/lib.rs (hunk 2 above)
Comment (buxx) [reply src/lib.rs]: done

File: examples/demo.rs (hunk 1 above)
Comment (buxx) [reply examples/demo.rs]: done

**Comment by buxx** [comment]
Hi emilk,

Modifications done :)

**Pull request merged**